pkg/ingester/ingester.go

```diff
@@ -1347,7 +1347,7 @@ func (i *Ingester) Push(ctx context.Context, req *cortexpb.WriteRequest) (*corte
 			return nil, wrapWithUser(err, userID)
 		}
 
-		if i.cfg.BlocksStorageConfig.TSDB.EnableNativeHistograms {
+		if i.limits.EnableNativeHistogramPerUser(userID) {
```

Comment:
I think we need to throw an exception here similar to Prometheus? Wdyt?


https://github.com/prometheus/prometheus/blob/main/tsdb/head_append.go#L647


Try ingesting some native histogram sample into current version of cortex with the NH feature disabled and see what is the behaviour.

Reply:
I think current Cortex behaviour is that it count is discarded sample here and not throw error here. But I m checking by running Cortex.